src/haddock/modules/__init__.py

```diff
@@ -297,6 +300,16 @@ def export_io_models(self, faulty_tolerance=0):
                 f"{faulty:.2f}% of output was not generated for this module "
                 f"and tolerance was set to {faulty_tolerance:.2f}%."
             )
+            # Try to detect CNS errors
+            if detected_errors := find_all_cns_errors(self.path):
+                _msg += linesep
+                for error in detected_errors.values():
+                    _msg += (
+                        # f'An error was detected in {len(error["files"])} files'
+                        # f'({",".join(error["files"][:3])}...).{linesep}'
+                        f'{str(error["error"])}{linesep}'
+                        )
+            # Show final error message
```

Comment:
This is also in the wrong place, need to raise this error in the base `CNSJob` class;

https://github.com/haddocking/haddock3/blob/a15dc5e7569d53421c861dcef4980149d6cc90d4/src/haddock/libs/libsubprocess.py#L93-L94

Reply:
I would keep it here, as we just want to check what errors are present when the workflow is about to terminate, and not after each CNS jobs.
So it limits the number of `.out` files that we read, and only parse them when models have not been generated